extensions/mapstore/src/main/java/com/hazelcast/mapstore/Queries.java

```diff
@@ -43,34 +43,31 @@ class Queries {
     private final Map<Integer, String> deleteAllQueries = new ConcurrentHashMap<>();
 
     Queries(String mapping, String idColumn, List<SqlColumnMetadata> columnMetadata) {
-        loadQuery = "SELECT * FROM \"" + mapping + "\" WHERE \"" + idColumn + "\" = ?";
+        loadQuery = String.format("SELECT * FROM \"%s\" WHERE \"%s\" = ?", mapping, idColumn);
```

Comment:
The values of these strings come from the XML / Yaml configuration. Do these really need to be escaped?

Calcite provides `org.apache.calcite.sql.SqlDialect#quoteIdentifier(java.lang.String)`, which escapes double quotes by doubling them. But I am not sure we should introduce a direct dependency on calcite here.

We don't care about different dialects here because this always runs against Hazelcast SQL.

Reply:
1. If we escape all identifiers, then MySQL needs to run in Ansi SQL mode.  With "sessionVariables=sql_mode=ANSI"  For example like this
docker run \
  -d \
  -it \ 
  -e MYSQL_ROOT_PASSWORD=root \
  -e MYSQL_DATABASE=db  \
  --name mysql \
  --rm \
  -p 3306:3306 \
  -v ${PWD}/schema.sql:/docker-entrypoint-initdb.d/schema.sql \
  mysql:8.0 --sql-mode="ANSI_QUOTES"

But most servers do not run with sql-mode ON. 

And for MySQL table and column escaping character is backtick not double quote. This works for normal servers even though column name has a hyphen (normally considered minus operator) 

CREATE TABLE foo (
  `time-taken` int(10) unsigned NOT NULL 
);

This does not **work** for a normal server
CREATE TABLE foo (
  "time-taken" int(10) unsigned NOT NULL 
);
So adding double quote for MySQL is not conventional. 

2. For PostgreSQL, the default behavior is converting everything to lower case.  If we use double quotes then PostgreSQL will use everything as it is. But the below example is going give error, because table name has to be used with double quotes all the time. And this is not the convention in PostgreSQL. People will get confused.
CREATE TABLE "People" (
 id SERIAL NOT NULL,
 email TEXT NOT NULL,
 PRIMARY KEY(id)
);

select * from people; <-- Error here

ERROR: relation "people" does not exist

See https://lerner.co.il/2013/11/30/quoting-postgresql/

3. Similar for MS SQL . Please see
https://learn.microsoft.com/en-us/sql/t-sql/statements/set-quoted-identifier-transact-sql?redirectedfrom=MSDN&view=sql-server-ver16

Bottom line : 
In my opinion we should get rid of all quotes around column and table names.

Reply:
Ok. I have used the SqlDialect class to quote the identifiers